src/script/common/c_converter.cpp

```diff
@@ -179,6 +179,12 @@ v3f check_v3f(lua_State *L, int index)
 	CHECK_POS_COORD("z");
 	pos.Z = lua_tonumber(L, -1);
 	lua_pop(L, 1);
+	if(util_isnan(pos.X))
+		throw LuaError("NaN passed as X coordinate");
+	if(util_isnan(pos.Y))
+		throw LuaError("NaN passed as Y coordinate");
+	if(util_isnan(pos.Z))
+		throw LuaError("NaN passed as Z coordinate");
```

Comment:
Integrating to CHECK_POS_COORD is probably a good idea. I'll get back to fixing this up after finishing a larger thing I'm working on.